src/components/UNSTABLE__Shell/HeaderAction/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file Header action.
+ * @copyright IBM Security 2020
+ */
+
+import classnames from 'classnames';
+import { bool, func, node, string } from 'prop-types';
+import React, { cloneElement, useRef } from 'react';
+
+import IconButton from '../../IconButton';
+import Transition from '../../Transition';
+
+import { namespace } from '../Shell';
+import { headerNamespace } from '../ShellHeader';
+
+import popoverNamespace from '../Popover';
+
+function HeaderAction({
+  activeAction,
+  children,
+  className,
+  hasBadge,
+  id,
+  popover,
+  setActiveAction,
+  ...other
+}) {
+  const isActive = popover && activeAction === id;
+
+  const ref = useRef(null);
+  const popoverRef = useRef(null);
+
+  const headerButtonNamespace = `${headerNamespace}__button`;
+  const isIconButton = popover && children.type === IconButton;
+
+  function onBlur({ relatedTarget }) {
+    return (
+      isActive &&
+      !popoverRef.current.contains(relatedTarget) &&
+      relatedTarget !== ref.current &&
+      setActiveAction(null)
+    );
+  }
+
+  return (
+    <>
+      {cloneElement(children, {
```

Comment:
Wondering if `cloneElement` is necessary here?

Reply:
Because the consumer can provide a number of things here, and we need to apply particular classes and handlers to manage the state between actions, I don't think there's any way of avoiding this. What's the alternative approach?